Add popup detail

ram1117 opened the pull request:
1.create markup,style and javascript functions for project pop up for  mobile version
2.create markup,style and javascript functions for project pop up  for desktop version
3.Clear eslinter and stylelint errors in the project
4.add comments in js files for clarity
5.update readme file with new changes in the project

ram1117 committed:
add javascript objects for projects,add new js files

ram1117 committed:
Refactor projects section in mobile and desktop versions

projects section project tiles created using javascript
split index.js file to multiple js files for modularity

ram1117 committed:
add popup js file to create mobile and desktop popups

ram1117 committed:
reorganize scripts and folders for javascript file

jossi30 committed:
added a popup to the project

ram1117 committed:
add popup menu for mobile version

add styling,markup using js for mobile popup menu
add two icon files for popup window

ram1117 committed:
add project scroll functionality to popup window

ram1117 committed:
add desktop popup styling and javascript functions

ram1117 committed:
add thumbnail scroll markup,style and js in popup window

ram1117 committed:
Clear eslint and stylint errors from project

ram1117 committed:
add comments to js file, update readme file

ram1117 committed:
clear one eslint error in index.js file

Favourezeugwa commented on scripts/index.js:
```diff
@@ -41,3 +41,349 @@ navListItem.forEach((listItem) => {
+  });
+  const projectButton = document.createElement('button');
+  projectButton.classList.add('button-generic', 'button-project');
+  projectButton.textContent = 'See this Project';
```
[OPTIONAL] 👍🏽 

- Great Job adding popup modal to your desktop version 👍🏽 

- I'd like to kindly suggest that you ensure that your button `see this project` appears on your mobile screen too inorder for us to improve user experience on mobile screen. 

This is what i can see compared to Figma 👇🏽 

| Your Design  | Figma Design |
| :---:   | :---: | 
| ![nosep](https://user-images.githubusercontent.com/99246359/208944139-160af4c4-55fc-46b7-9d50-d5abfcbc54ad.PNG) |![figm](https://user-images.githubusercontent.com/99246359/208944675-c999bc36-5b05-485e-83c1-d9d99a25b78f.PNG)  |

ram1117 replied:
@Favourezeugwa Hi Favour, I have attached a screenshot of how this part looks in my browser for a screen approximately 550px wide. Can i know at what resolution you were testing this please. I cannot reproduce this in my browser(firefox).

![Screenshot_20221221_215057](https://user-images.githubusercontent.com/116718155/208954525-b4d00973-a565-4c52-8c5c-b3d893ed5a69.png)

Favourezeugwa replied:
That's alright @ram1117, I used 375px (ideal screen size for mobile screens). I guess there was a bug. I can see the `see this project` button now. This will be marked as optional. You can go ahead with fixing the rest changes 🌸

jerryowusu replied:
> @Favourezeugwa Hi Favour, I have attached a screenshot of how this part looks in my browser for a screen approximately 550px wide. Can i know at what resolution you were testing this please. I cannot reproduce this in my browser(firefox).
> 
> ![Screenshot_20221221_215057](https://user-images.githubusercontent.com/116718155/208954525-b4d00973-a565-4c52-8c5c-b3d893ed5a69.png)

Hello @ram1117 and @Favourezeugwa, I believe what @Favourezeugwa was right on making that observation because of the way @ram1117 is creating the modal. The desktop and the mobile modals are being created separately hence it is not working responsively. If you open the modal popup from a desktop screen width it maintains the same shape and dimensions without switching to the mobile popup modal for all screens of both desktop and mobile and vice versa.

@ram1117, kindly check and rectify this issue in the index.js file, lines 299 - 311 `createPopup` function. :wink:

Favourezeugwa commented on scripts/index.js:
```diff
@@ -41,3 +41,349 @@ navListItem.forEach((listItem) => {
+}
+
+// function to create popup for mobile version
+function createMobilePopup() {
```
- I think the popup modal on your mobile screen is a bit too big, can we fix it?? Lets try as much as possible to make the design look like figma. I guess you may have forgotten to style your popup for mobile screen, hence the sizes. 

Here is what I mean 👇🏽 

| Your Design  |That of Figma Design |
| :---:   | :---: | 
| ![mobp](https://user-images.githubusercontent.com/99246359/208946574-c3e91e1b-6efb-4680-b53a-3a9eaf119b1b.PNG) |![mobpop](https://user-images.githubusercontent.com/99246359/208946930-25c78b04-1bb9-437a-8fb9-b0dd80c02b5a.PNG)  |

ram1117 replied:
@Favourezeugwa hi. I have attached below, how the page looks in my browser at approx 550px wide. could you please share the browser and resolution you are testing, i couldn't recreate this in my browser.
![Screenshot_20221221_215258](https://user-images.githubusercontent.com/116718155/208955462-5f9c2556-2d57-4f58-8c83-f38c7aa2d4dd.png)

Favourezeugwa replied:
- Okay @ram1117 , For mobile sizes, let's stick with 375px.  In this review comment, kindly fix just the boxes containing the `Ruby on rails`, `HTML/CSS`, and `JavaScript`.

Here is what I mean 👇🏽 
| The technologies on mobile screen  |
| :---:   | 
|![nefx](https://user-images.githubusercontent.com/99246359/208967024-6c5b0e33-9767-46e5-b280-0a915ded6fe1.PNG)  |


- They appear to be a bit too large 🤗

ram1117 replied:
@Favourezeugwa  hi Favoure, thank you for clarifying. I could see the problem and I am working on fixing this.

Favourezeugwa commented on scripts/index.js:
```diff
@@ -41,3 +41,349 @@ navListItem.forEach((listItem) => {
+});
+
+// function to load project details into popup
+function loadNewProj(projectObj) {
+  const pheader = document.querySelector('.popup-header');
+  const ptechCntnr = document.querySelector('.pop-tech-cntnr');
+  const pscrollCntnr = document.querySelector('.project-scroll-container');
+  const thmbScroll = document.querySelector('.thumbnail-scroll');
+  const lButton = document.querySelector('.left-button');
+  const rButton = document.querySelector('.right-button');
+  const pdescription = document.querySelector('.popup-description');
+  const demoLink = document.querySelector('.live-button');
+  const srcLink = document.querySelector('.src-button');
```
- I think it will be great if we added some space between the text and images on your desktop's popup modal. 

Here is what I mean 👇🏽 

| Your Popup |
| :---:   | 
|![spc](https://user-images.githubusercontent.com/99246359/208948207-974b4c86-d6fe-48f2-8750-dcece36735b5.PNG)  |

|Figma's Popup |
| :---:   | 
|![spcgd](https://user-images.githubusercontent.com/99246359/208948279-8e6f6c0b-5914-41b7-9c11-bb57fab21314.PNG)  |


- Also, let's kindly remember to include the dark blue background color that's just at the edges of each popup(desktop).  🤗 

Here is what I mean 👇🏽 

|Figma's Darkblue edges |
| :---:   | 
|![drkgr](https://user-images.githubusercontent.com/99246359/208948947-a1bfe9de-b5b7-4ccb-b020-455e6e2138c5.PNG)  |

|Yours (has the whitespace color) |
| :---:   | 
|![yus](https://user-images.githubusercontent.com/99246359/208949401-dde451cf-3370-4b84-adfb-51924129f497.PNG)  |

Favourezeugwa reviewed (changes requested):
## STATUS: REQUIRES CHANGES 🚦 ⚠️ ♻️ 

Hi @ram1117  , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


![close lady](https://user-images.githubusercontent.com/99246359/208949964-6c8c38af-d96f-4905-88f3-44204658e88c.gif)


## Highlights 💯 
- No Linter errors ✔️
- Descriptive Pull request summary✔️
- Good commit messages ✔️
- Professional Readme file ✔️ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

ram1117 committed:
Implement code review 1 comments

1.added spacing in pop window desktop
2.added border to thumbnail in pop up window
3.added modal window border in desktop version

ram1117 committed:
fix linter error after code review 1 changes

ram1117 committed:
modify style.css to change the font size of technology items in mobile popup

jerryowusu commented on scripts/index.js:
```diff
@@ -41,3 +41,366 @@ navListItem.forEach((listItem) => {
+  return desktopModal;
+}
+
+function createPopup(project) {
+  let popupwindow;
+  if (window.innerWidth <= 768) {
+    popupwindow = createMobilePopup();
+  }
+  if (window.innerWidth > 768) {
+    popupwindow = createDesktopPopup();
+  }
+  const container = document.querySelector('.projects-container');
+  container.appendChild(popupwindow);
+  loadNewProj(project);
+  document.body.style.overflow = 'hidden';
+}
```
As requested by the previous reviewer, kindly ensure that your design matches that of the figma design. The skills  boxes containing the `Ruby on rails`, `HTML/CSS`, and `JavaScript` looks bigger than the actual design. :confused: This is because the desktop and the mobile modals are being created separately hence it is not working responsively and the media queries from the stylesheet is not working. If you open the modal popup from a desktop screen width it maintains the same shape and dimensions without switching to the mobile popup modal for all screens of both desktop and mobile and vice versa. :wink: 
Kindly ensure that you create only one modal popup and use css media queries to switch from mobile to desktop and vice versa instead of creating two popup windows. :+1: 

### Mobile version when created from lines 304-306 (index.js)

![Screenshot from 2022-12-21 19-01-59](https://user-images.githubusercontent.com/65233860/208986014-f834ac66-ba72-4ed4-9ac6-0194990374cf.png)

### Actual design 

![Screenshot from 2022-12-21 19-25-48](https://user-images.githubusercontent.com/65233860/208986985-7d3a3b13-cae1-4150-8a05-8c10c5ffe827.png)




.....................................................................................................................................................

## Please kindly check these other images for optional perspective

### Mobile version when created from lines 304-306 (index.js)

![Screenshot from 2022-12-21 19-02-21](https://user-images.githubusercontent.com/65233860/208986034-3b0c11d1-a8d1-41f0-8561-b8e65b85a749.png)


### Desktop version If created from lines 300-303 (index.js)

![Screenshot from 2022-12-21 19-02-58](https://user-images.githubusercontent.com/65233860/208986058-8c066fd1-1f9f-4731-8eb3-60850873ea97.png)

jerryowusu reviewed (changes requested):
Hi @ram1117 , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

![so-close](https://user-images.githubusercontent.com/65233860/208987839-1b100c51-48f9-4896-9320-929ece4388e8.gif)


 ## Highlights
All linters are passing :heavy_check_mark: 
Great effort so far :heavy_check_mark: 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

ram1117 committed:
modify popup window design based on code review 2 comments

ram1117 committed:
modify styling in popup window to match figma design both mobile and desktop version

Gopxfs reviewed (approved):
# 🟢 STATUS: APPROVED 🟢

## Hello @ram1117 👋,

Your project is complete! You followed all the requirements, feel free to merge it! 😄
Congratulations! 🎉 

<img src="https://media.giphy.com/media/cXblnKXr2BQOaYnTni/giphy.gif">

## Highlights
✔️ Linters are passing successfully.
✔️ Implemented all the changes requested by the previous reviewer. Well done!

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
